ui/app/AppLayouts/Communities/panels/SharedAddressesPanel.qml

```diff
@@ -0,0 +1,680 @@
+import QtQuick 2.15
+import QtQml.Models 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+import QtGraphicalEffects 1.15
+
+import StatusQ.Core 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Core.Utils 0.1
+
+import SortFilterProxyModel 0.2
+
+import AppLayouts.Profile.controls 1.0
+import AppLayouts.Communities.controls 1.0
+import AppLayouts.Communities.views 1.0
+import AppLayouts.Communities.helpers 1.0
+
+import utils 1.0
+
+Control {
+    id: root
+
+    property bool isEditMode
+
+    required property string communityName
+    required property string communityIcon
+    property int loginType: Constants.LoginType.Password
+
+    required property var walletAccountsModel // name, address, emoji, colorId, assets
+    required property var permissionsModel  // id, key, permissionType, holdingsListModel, channelsListModel, isPrivate, tokenCriteriaMet
+    required property var assetsModel
+    required property var collectiblesModel
+
+    readonly property string title: isEditMode ? qsTr("Edit which addresses you share with %1").arg(communityName)
+                                               : qsTr("Select addresses to share with %1").arg(communityName)
+
+    readonly property var buttons: ObjectModel {
+        StatusFlatButton {
+            visible: root.isEditMode
+            borderColor: Theme.palette.baseColor2
+            text: qsTr("Cancel")
+            onClicked: root.close()
+        }
+        StatusButton {
+            visible: root.isEditMode
+            icon.name: Constants.authenticationIconByType[root.loginType]
+            text: qsTr("Save changes")
+            onClicked: {
+                // TODO connect to backend
+                root.close()
+            }
+        }
+        StatusButton {
+            visible: !root.isEditMode
+            enabled: root.selectedAirdropAddress && root.selectedSharedAddresses.length
+            text: qsTr("Share selected addresses to join")
+            onClicked: {
+                root.shareSelectedAddressesClicked(root.selectedAirdropAddress, root.selectedSharedAddresses)
+                root.close()
+            }
+        }
+        // NB no more buttons after this, see property `rightButtons` below
+    }
+
+    readonly property var rightButtons: [buttons.get(buttons.count-1)] // "magically" used by CommunityIntroDialog StatusStackModal impl
```

Comment:
Ohh, this "magic" is really hard to spot..

Reply:
That's why I also added the comment above (line 64)